server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesResponse.java

```diff
@@ -75,5 +126,158 @@ public void writeTo(StreamOutput out) throws IOException {
+                    builder.startObject(entry.key);
+                    {
+                        builder.startObject("aliases");
+                        {
+                            for (final AliasMetaData alias : entry.value) {
+                                AliasMetaData.Builder.toXContent(alias, builder, ToXContent.EMPTY_PARAMS);
+                            }
+                        }
+                        builder.endObject();
+                    }
+                    builder.endObject();
+                }
+            }
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static GetAliasesResponse fromXContent(XContentParser parser) throws IOException {
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+        }
+        ensureExpectedToken(Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> aliasesBuilder = ImmutableOpenMap.builder();
+
+        String currentFieldName;
+        Token token;
+        String exceptionMsg = null;
+        RestStatus status = RestStatus.OK;
+        ElasticsearchException exception = null;
+
+        while ((token = parser.nextToken()) != Token.END_OBJECT) {
+            if (parser.currentToken() == Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+
+                if ("status".equals(currentFieldName)) {
+                    if ((token = parser.nextToken()) != Token.FIELD_NAME) {
+                        ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, parser::getTokenLocation);
+                        status = RestStatus.fromCode(parser.intValue());
+                    }
+                } else if ("error".equals(currentFieldName)) {
+                    if ((token = parser.nextToken()) != Token.FIELD_NAME) {
+                        if (token == Token.VALUE_STRING) {
+                            exceptionMsg = parser.text();
+                        } else if (token == Token.START_OBJECT) {
+                            token = parser.nextToken();
+                            exception = ElasticsearchException.innerFromXContent(parser, true);
+                        }
+                    }
+                } else {
+                    String indexName = parser.currentName();
+                    if (parser.nextToken() == Token.START_OBJECT) {
+                        List<AliasMetaData> parseInside = parseAliases(parser);
+                        aliasesBuilder.put(indexName, parseInside);
+                    }
+                }
+            }
+        }
+        if (exception != null) {
+            throw new ElasticsearchStatusException(exception.getMessage(), status, exception.getCause());
+        }
+        if (RestStatus.OK != status && aliasesBuilder.isEmpty()) {
+            throw new ElasticsearchStatusException(exceptionMsg, status);
+        }
```

Comment:
shouldn't this one already be covered by the previous `if`?

Reply:
The previous `if` is covering the case when an ES exception has been parsed ( a concrete index was not found ). This `if` is covering the case, when a string exception has been generated ( the concrete alias could not be found ).